README
Wammu
=====

GUI for Gammu library.

Homepage
========

<http://wammu.eu/>

License
=======

GNU GPL version 2.

First start
===========

On first start you will be asked for setting up phone parameter. If you never
used Gammu/Wammu before, phone searching will be suggested, which should do
the job for you.

Usage
=====

First you have to connect to phone, then you can perform some operations with
it. For creating entries and importing you do not need to read things from
phone, for others you have to (surprising? :-)).

All actions with current list are accessible from context menu on each item,
you can also use keys: Enter for editing and Delete for deleting.

Backup from main menu creates backup of entries you have already retrieved
from phone.

Bug reporting
=============

Please report found bugs to <http://bugs.wammu.eu>.

Translating
===========

You can help translating Wammu to your language on translation server -
<https://l10n.cihar.com/projects/wammu/>.

Version control
===============

Developers might be interested in snapshot or version control system
(Subversion is being used), information about accessing these is
available at <http://wammu.eu/download/>.

Anonymous Subversion is available using <svn://svn.cihar.com/wammu>,
there is also web browser for it - <http://viewsvn.cihar.com/wammu>.

# vim: et ts=4 sw=4 sts=4 tw=72 spell spelllang=en_us


NEWS
NEWS
====

Major changes for each release are described here. Detailed changes between
two versions can be retrieved from Subversion.

0.26
==== (2008-??-??)

* Various fixes to make Windows version back to usable state.

0.25
==== (2007-12-19)

* Fall back to Add methods if Set is not supported.
* Fixed phone number validation.
* Fixed crash when org.freedesktop.Notifications service is not
  provided.
* Correctly report why guided wizard can not connect to phone.
* New navigation icons, thanks to David Corrales.

0.24
==== (2007-11-13)

* Notifies user about incoming calls using D-Bus.
* Remember default type of created entry (eg. phone/SIM memory).
* Newly created entries contain some reasonable fields by default.
* Fixed validation of editor fields.
* Improvements in guided wizard to make it easier to use.
* Clever searching over Bluetooth based on manufacturer.
* Searching can be really limited to some connection types.

0.23
==== (2007-10-05)

* Fixed phone number prefix settings.
* Customizable format for displaying name.
* Fixed crash when reporting information to phone database.

0.22
==== (2007-09-11)

* Fixed errors in locales and applied stricter checking.
* Fixed various crashes.

0.21
==== (2007-06-20)

* Tell user if he needs to be member of some group while searching.
* Avoid blocking opening of a browser.

0.20
==== (2007-05-16)

* Improve locales behavior (and require wxPython with Unicode).
* Show time in call listings.
* Fixed infinite loop in searching.
* Fixed displaying of messages with special chars.
* Fixed locale detection on system without configured locales.
* Use different approach for detecting home directory on Windows.
* Now requires PyWin32 on Windows.

0.19
==== (2007-03-11)

* Improved SMS recipient list handling.
* Can save and load SMS recipient list.
* Fix crash on Windows when username contains non-ascii chars.
* Fix Windows binary crashes.
* Support for sending file to phone.
* Support for task bar icon (thanks to Rene Peters).
* Fixed debug log on Windows.
* Fixed error handler on Windows.

0.18
==== (2007-02-24)

* Better setup.py to use more of distutils.
* .mo locales are generated on build and not included in VCS.
* Show only existing devices on Windows (this requires pywin32).
* Now requires wxPython at least 2.6.2.
* Support for reporting in Gammu Phone Database.
* Automatically asks user for feedback after month of usage.
* GUI more follows HIG on target platform (button placement).
* Add wildcard search possibility.
* Use gammu configuration for phone connection instead of own.

0.17
==== (2006-10-19)

* Fix searching.
* Remember column sorting.

0.16
==== (2006-09-21)

* Include platform in crash log.
* Fix SMS composer on Windows.
* Fix adding part to SMS after removing all.
* Fix editor on Windows.
* Now requires wxWidgets 2.6.

0.15
==== (2006-09-03)

* Language selection for Windows (based on GTK+ 2 ISS code).
* Finnish translation from Rosetta.
* Searching can be done plain text and not only regexp.
* Phone number matching now works even for national numbers.
* Fixed usability with inversed themes.

0.14
==== (2006-08-05)

* Automatically append backup extension.
* PyBluez is now default for Bluetooth devices lookup.
* Fixed mail export for non English locales.
* Fixed installation on Windows.
* Added support for py2exe for distributing Windows executable.
* Added script for creating Wammu installer on Windows (using Inno Setup).

0.13
==== (2006-03-24)

* Possibility to export SMS to maildir, mailbox and IMAP account.
* More user friendly debug logging.
* Many new translations, thanks to Rosetta.
* Improved Bluetooth lookup - now uses either PyBluez or btctl (part of GNOME
  Bluetooth).

0.12
==== (2006-01-08)

* More user friendly reporting of some common errors.
* Many bug fixes.

0.11
==== (2005-12-24)

New features:

* SMS backups and import support.

0.10
==== (2005-08-07)

Bug fixes:

* Fixed wrong implementation of not synced items for SMS.
* Some minor display errors were fixed.
* Application icon is now used correctly.

0.9
=== (2005-07-24)

New features:

* Searching in entries.
* Reading backups without phone connected, can work as backup viewer.
* New bug reporting dialog, added bugid to allow easily search for
  similar bugs.
* Support for searching Bluetooth devices using gnomebt python module.
* Better searching for Nokia phones.

Bug fixes:

* Fixed some issues with debug wxPython, probably still all not solved.

0.8
=== (2004-09-17)

New features:

* New settings dialog.
* Better date editing control.

Bug fixes:

* Should work with both wxPython 2.5 and 2.4.

Known issues:

* Calendar editor does not work in wxPython 2.5/Gtk 2, you need to enter
  values manually (this is wxWidgets issue and has been fixed there).

0.7
=== (2004-09-06)

New features:

* Signal and battery state continuous monitoring.
* Support for sending messages to multiple recipients.

Bug fixes:

* Should work with wxPython 2.5.

0.6
=== (2004-03-03)

New features:

* Can expand "compressed" SMSes - TextCOMPRESSIONlikeTHIS.
* Allow direct calls/send message for contacts.

Bug fixes:

* Lots of charset issues fixed.

0.5.1
===== (2004-01-23)

* Fixed typo in Wammu/Utils.py.

0.5
=== (2004-01-21)

New features:

* Searching for phone.
* Backup and import functionality.
* Deleting of multiple entries at once.
* Reply on message.
* Playback of ring tones using timidity.
* More decoded SMS parts are now recognized (especially Nokia ones were not
  recognized before).
* Unexpected exceptions are handled and reported as bugs.

Wammu/Data.py
# -*- coding: UTF-8 -*-
# vim: expandtab sw=4 ts=4 sts=4:
'''
Wammu - Phone manager
Some static data like bitmaps, category mappings etc.
Many of them might be moved to python-gammu later
'''
__author__ = 'Michal Čihař'
__email__ = 'michal@cihar.com'
__license__ = '''
Copyright (c) 2003 - 2007 Michal Čihař

This program is free software; you can redistribute it and/or modify it
under the terms of the GNU General Public License version 2 as published by
the Free Software Foundation.

This program is distributed in the hope that it will be useful, but WITHOUT
ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
more details.

You should have received a copy of the GNU General Public License along with
this program; if not, write to the Free Software Foundation, Inc.,
51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
'''

import sys
from gammu.Data import Connections, MemoryValueTypes, CalendarTypes, CalendarValueTypes, TodoPriorities, TodoValueTypes, InternationalPrefixes


# When support for sound will be implemented, here should be sounds
PredefinedSounds = [
        (_('Chimes high'),''),
        (_('Chimes low'),''),
        (_('Ding'),''),
        (_('TaDa'),''),
        (_('Notify'),''),
        (_('Drum'),''),
        (_('Claps'),''),
        (_('Fanfare'),''),
        (_('Chord high'),''),
        (_('Chord low'),''),
        ]

# Wanted somebody who will draw nicer icons :-)

Note = [
    '16 16 2 1',
    'x c Black',
    '  c None',
    '      xx        ',
    '      xxx  xxx  ',
    '      xxxxxxxxx ',
    '      xx  xx  xx',
    '      xx        ',
    '      xx        ',
    '      xx        ',
    '      xx        ',
    '  xxxxxx        ',
    ' xxxxxxx        ',
    'xxxxxxxx        ',
    'xxxxxxxx        ',
    'x  xxxxx        ',
    'x   xxxx        ',
    ' x  xxx         ',
    '  xxxx          ']

UnknownPredefined = [
    '16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxxxx     ',
    '    x      x    ',
    '   x  xxxx  x   ',
    '   x x    x x   ',
    '    x     x x   ',
    '          x x   ',
    '         x x    ',
    '        x x     ',
    '       x x      ',
    '       x x      ',
    '       xx       ',
    '                ',
    '       xx       ',
    '      xxxx      ',
    '      x xx      ',
    '       xx       ']

PredefinedAnimations = [
    (_("I'm ironic, flirty"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxxxx     ',
    '   xx      xx   ',
    '  x          x  ',
    ' x            x ',
    ' x  xx   xx   x ',
    'x   x x  x x   x',
    'x    xx   xx   x',
    'x              x',
    'x              x',
    'x              x',
    'x   x      x   x',
    ' x   x    x   x ',
    ' x    x  x    x ',
    '  x    xx    x  ',
    '   xx      xx   ',
    '     xxxxxx     ']),
    (_("I am glad"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxxxx     ',
    '   xx      xx   ',
    '  x          x  ',
    ' x            x ',
    ' x  xx   xx   x ',
    'x   x x  x x   x',
    'x    xx   xx   x',
    'x              x',
    'x              x',
    'x              x',
    'x   x      x   x',
    ' x   x    x   x ',
    ' x    xxxx    x ',
    '  x          x  ',
    '   xx      xx   ',
    '     xxxxxx     ']),
    (_("I am skeptic"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxxxx     ',
    '   xx      xx   ',
    '  x          x  ',
    ' x            x ',
    ' x  xx   xx   x ',
    'x   x x  x x   x',
    'x    xx   xx   x',
    'x              x',
    'x              x',
    'x              x',
    'x   xxxxx      x',
    ' x       x    x ',
    ' x        x   x ',
    '  x          x  ',
    '   xx      xx   ',
    '     xxxxxx     ']),
    (_("I am sad"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxxxx     ',
    '   xx      xx   ',
    '  x          x  ',
    ' x            x ',
    ' x  xx   xx   x ',
    'x   x x  x x   x',
    'x    xx   xx   x',
    'x              x',
    'x              x',
    'x              x',
    'x     xxxx     x',
    ' x   x    x   x ',
    ' x  x      x  x ',
    '  x          x  ',
    '   xx      xx   ',
    '     xxxxxx     ']),
    (_("WOW"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxxxx     ',
    '   xx      xx   ',
    '  x          x  ',
    ' x            x ',
    ' x  xx   xx   x ',
    'x   x x  x x   x',
    'x    xx   xx   x',
    'x              x',
    'x              x',
    'x     xxxx     x',
    'x    x    x    x',
    ' x   x    x   x ',
    ' x    xxxx    x ',
    '  x          x  ',
    '   xx      xx   ',
    '     xxxxxx     ']),
    (_("I am crying"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxxxx     ',
    '   xx      xx   ',
    '  x          x  ',
    ' x            x ',
    ' x  xx   xx   x ',
    'x   x x  x x   x',
    'x    xx   xx   x',
    'x     x    x   x',
    'x              x',
    'x              x',
    'x     xxxx     x',
    ' x   x    x   x ',
    ' x    xxxx    x ',
    '  x          x  ',
    '   xx      xx   ',
    '     xxxxxx     ']),
    (_("I am winking"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxxxx     ',
    '   xx      xx   ',
    '  x          x  ',
    ' x            x ',
    ' x       xx   x ',
    'x  xxxx  x x   x',
    'x    xx   xx   x',
    'x              x',
    'x              x',
    'x              x',
    'x   x      x   x',
    ' x   xxxxxx   x ',
    ' x            x ',
    '  x          x  ',
    '   xx      xx   ',
    '     xxxxxx     ']),
    (_("I am laughing"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxxxx     ',
    '   xx      xx   ',
    '  x          x  ',
    ' x            x ',
    ' x  xx   xx   x ',
    'x   x x  x x   x',
    'x    xx   xx   x',
    'x              x',
    'x              x',
    'x              x',
    'x   x      x   x',
    ' x   xxxxxx   x ',
    ' x    xxxx    x ',
    '  x          x  ',
    '   xx      xx   ',
    '     xxxxxx     ']),
    (_("I am indifferent"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxxxx     ',
    '   xx      xx   ',
    '  x          x  ',
    ' x            x ',
    ' x  xx   xx   x ',
    'x   x x  x x   x',
    'x    xx   xx   x',
    'x              x',
    'x              x',
    'x              x',
    'x              x',
    ' x            x ',
    ' x   xxxxxx   x ',
    '  x          x  ',
    '   xx      xx   ',
    '     xxxxxx     ']),
    (_("I am in love"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxx xx xx ',
    '   xx    x  x  x',
    '  x       x   x ',
    ' x         x x  ',
    ' x  xx   xx x x ',
    'x   x x  x x   x',
    'x    xx   xx   x',
    'x              x',
    'x              x',
    'x              x',
    'x              x',
    ' x   x    x   x ',
    ' x    xxxx    x ',
    '  x          x  ',
    '   xx      xx   ',
    '     xxxxxx     ']),
    (_("I am confused"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxxxx xx  ',
    '   xx      x  x ',
    '  x           x ',
    ' x           x  ',
    ' x  xx   xx    x',
    'x   x x  x x x x',
    'x    xx   xx   x',
    'x              x',
    'x              x',
    'x              x',
    'x              x',
    ' x   xxxxx    x ',
    ' x            x ',
    '  x          x  ',
    '   xx      xx   ',
    '     xxxxxx     ']),
    (_("Tongue hanging out"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxxxx     ',
    '   xx      xx   ',
    '  x          x  ',
    ' x            x ',
    ' x  xx   xx   x ',
    'x   x x  x x   x',
    'x    xx   xx   x',
    'x              x',
    'x              x',
    'x              x',
    'x              x',
    ' x         x  x ',
    ' x    xxxxx x x ',
    '  x     xx   x  ',
    '   xx     x   x ',
    '     xxxxxxxxx  ']),
    (_("I am angry"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxxxx     ',
    '   xx      xx   ',
    '  x          x  ',
    ' x x        x x ',
    ' x  xx   xxx  x ',
    'x   x x  x x   x',
    'x    xx   xx   x',
    'x              x',
    'x              x',
    'x              x',
    'x     xxxx     x',
    ' x   x    x   x ',
    ' x  x      x  x ',
    '  x          x  ',
    '   xx      xx   ',
    '     xxxxxx     ']),
    (_("Wearing glases"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    '     xxxxxx     ',
    '   xx      xx   ',
    '  x          x  ',
    ' x            x ',
    'xxxxxxxxxxxxxxxx',
    'x   xxx  xxx   x',
    'x    xx   xx   x',
    'x              x',
    'x              x',
    'x              x',
    'x              x',
    ' x  x      x  x ',
    ' x   xxxxxx   x ',
    '  x          x  ',
    '   xx      xx   ',
    '     xxxxxx     ']),
    (_("Devil"),
    ['16 16 2 1',
    'x c Black',
    '  c None',
    'x    xxxxxx    x',
    'xxxxx      xxxxx',
    ' xx          xx ',
    ' x            x ',
    ' x  xx   xx   x ',
    'x   x x  x x   x',
    'x    xx   xx   x',
    'x              x',
    'x              x',
    'x   xxxxxxxx   x',
    'x   x x  x x   x',
    ' x   x    x   x ',
    ' x    xxxx    x ',
    '  x          x  ',
    '   xx      xx   ',
    '     xxxxxx     ']),
    ]

# First is used as default
Models = [
    'auto',
    'at',
    'alcatel',
    'nauto',
    'obex',
    'seobex',
    ]

Conn_Generic = [
    'at19200',
    'at115200',
    'obex',
    ]
Conn_Cable = [
    'at19200',
    'at115200',
    'fbusdlr3',
    'fbus',
    'mbus',
    'fbuspl2303',
    'phonetblue',
    'fbusblue',
    ]
Conn_IrDA_Win = [
    'irdaphonet',
    'irdaat',
    ]
Conn_IrDA_Other = [
    'irdaphonet',
    'at19200',
    ]
Conn_BlueRF = [
    'at19200',
    ]
Conn_Bluetooth_All = [
    'bluephonet',
    'bluefbus',
    'bluerfgnapbus',
    'blueat',
    'blueobex',
    'bluerfobex',
    'bluerfphonet',
    'bluerffbus',
    'bluerfat',
    ]
Conn_Bluetooth_Nokia = [
    'bluephonet',
    'bluefbus',
    'bluerfgnapbus',
    'bluerfphonet',
    'bluerffbus',
    'blueat',
    'bluerfat',
    'blueobex',
    'bluerfobex',
    ]
Conn_Bluetooth_Standard = [
    'blueat',
    'blueobex',
    'bluerfgnapbus',
    ]
Conn_Bluetooth = {
    'Sony-Ericsson' : Conn_Bluetooth_Standard,
    'Siemens' : Conn_Bluetooth_Standard,
    'BenQ' : Conn_Bluetooth_Standard,
    'Samsung' : Conn_Bluetooth_Standard,
    'LG' : Conn_Bluetooth_Standard,
    'Motorola' : Conn_Bluetooth_Standard,
    'Nokia' : Conn_Bluetooth_Nokia,
    'Alcatel' : Conn_Bluetooth_Standard,
    }
if sys.platform == 'win32':
    Devices = [
        '',
        'com1:',
        'com2:',
        'com3:',
        'com4:',
        'com5:',
        'com6:',
        ]
    AllDevices = [
        (Conn_IrDA_Win, '', None, ['irda']),
        (Conn_Cable, 'com%d:', (1,6), ['irda', 'usb', 'serial', 'bluetooth']),
        ]
# FIXME: support more platforms?
else:
    Devices = [
        '/dev/ttyS0',
        '/dev/ttyS1',
        '/dev/ttyUSB0',
        '/dev/ttyUSB1',
        '/dev/ttyACM0',
        '/dev/ttyACM1',
        '/dev/ircomm0',
        '/dev/rfcomm0',
        '/dev/usb/tts/0',
        ]
    AllDevices = [
        (Conn_Cable, '/dev/ttyS%d', (0, 3), ['serial']),
        (Conn_Cable, '/dev/ttyUSB%d', (0, 3), ['serial', 'usb']),
        (Conn_Cable, '/dev/ttyACM%d', (0, 3), ['serial', 'usb']),
        (Conn_BlueRF, '/dev/rfrcomm%d', (0, 1), ['bluetooth']),
        (Conn_IrDA_Other, '/dev/ircomm%d', (0, 1), ['irda']),
        (Conn_Cable, '/dev/usb/tts/%d', (0, 3), ['serial', 'usb']),
        ]

ContactMemoryTypes = ['ME', 'SM']

SMSIDs = {
    'Text':                 [
        'Text',
        'ConcatenatedTextLong',
        'ConcatenatedAutoTextLong',
        'ConcatenatedTextLong16bit',
        'ConcatenatedAutoTextLong16bit',
        'NokiaVCARD21Long',
        'NokiaVCALENDAR10Long'
        ],
    'Sound':                [
        'NokiaProfileLong',
        'NokiaRingtone',
        'NokiaRingtoneLong',
        'EMSSound10',
        'EMSSound12',
        'EMSSonyEricssonSound',
        'EMSSound10Long',
        'EMSSound12Long',
        'EMSSonyEricssonSoundLong',
        ],
    'Animation':            [
        'NokiaProfileLong',
        'EMSAnimation',
        'AlcatelMonoAnimationLong',
        'NokiaScreenSaverLong',
        ],
    'File':                 [
        'SiemensFile',
        ],
    'Bitmap':               [
        'NokiaProfileLong',
        'NokiaPictureImageLong',
        'NokiaOperatorLogo',
        'NokiaOperatorLogoLong',
        'NokiaCallerLogo',
        'EMSFixedBitmap',
        'EMSVariableBitmap',
        'EMSVariableBitmapLong',
        'AlcatelMonoBitmapLong',
        'AlcatelSMSTemplateName',
        ],
    'PredefinedAnimation':  [
        'EMSPredefinedAnimation',
        ],
    'PredefinedSound':      [
        'EMSPredefinedSound',
        ],
    }

TextFormats = [
    [(_('Alignment'), _('None')),
        ('Left', _('Left'), '<div align="left">%s</div>'),
        ('Right', _('Right'), '<div align="right">%s</div>'),
        ('Center', _('Center'), '<div align="center">%s</div>'),
        ],
    [(_('Text Size'), _('Normal')),
        ('Large', _('Large'), '<font size="+2">%s</font>'),
        ('Small', _('Small'), '<font size="-2">%s</font>'),
        ],
    ['', ('Bold', _('Bold'), '<b>%s</b>')],
    ['', ('Italic', _('Italic'), '<i>%s</i>')],
    ['', ('Underlined', _('Underlined'), '<u>%s</u>')],
    ['', ('Strikethrough', _('Strikethrough'), '<strike>%s</strike>')],
]

# dump from Gammu Phone Database
ManufacturerMap = {
    'Alcatel': 1,
    'Nokia': 2,
    'Siemens': 3,
    'Sony Ericsson': 4,
    'Sagem': 5,
    'Motorola': 6,
    'Falcom': 7,
    'Samsung': 8,
    'LG': 9,
    'Sharp': 10,
    'Mitsubishi': 11,
    'PalmOne': 12,
}
GarbleMap = {
    0: 'atdot',
    1: 'nospam',
    2: 'none',
    3: 'hide',
}

# Generated from http://standards.ieee.org/regauth/oui/oui.txt
MAC_Prefixes = {
        'Sony-Ericsson' : [
            '00:01:EC',
            '00:0A:D9',
            '00:0E:07',
            '00:0F:DE',
            '00:12:EE',
            '00:15:E0',
            '00:16:20',
            '00:16:B8',
            '00:18:13',
            '00:19:63',
            '00:1A:75',
            '00:1B:59',
            '00:1C:A4',
            '00:1D:28',
            '00:1E:45',
            '00:80:37',
            ],
        'Nokia' : [
            '00:02:EE',
            '00:0B:E1',
            '00:0E:ED',
            '00:0F:BB',
            '00:10:B3',
            '00:11:9F',
            '00:12:62',
            '00:13:70',
            '00:13:FD',
            '00:14:A7',
            '00:15:2A',
            '00:15:A0',
            '00:15:DE',
            '00:16:4E',
            '00:16:BC',
            '00:17:4B',
            '00:17:B0',
            '00:18:0F',
            '00:18:42',
            '00:18:8D',
            '00:18:C5',
            '00:19:2D',
            '00:19:4F',
            '00:19:79',
            '00:19:B7',
            '00:1A:16',
            '00:1A:89',
            '00:1A:DC',
            '00:1B:33',
            '00:1B:AF',
            '00:1B:EE',
            '00:1C:35',
            '00:1C:9A',
            '00:1C:D4',
            '00:1C:D6',
            '00:1D:3B',
            '00:1D:6E',
            '00:1D:98',
            '00:1D:E9',
            '00:1D:FD',
            '00:1E:3A',
            '00:1E:3B',
            '00:1E:A3',
            '00:1E:A4',
            '00:40:43',
            '00:A0:8E',
            '00:E0:03',
            ],
        'Siemens' : [
            '00:01:E3',
            '00:05:19',
            '00:0B:23',
            '00:0B:A3',
            '00:0D:41',
            '00:0E:8C',
            '00:0F:BB',
            '00:11:06',
            '00:11:33',
            '00:13:A3',
            '00:18:D1',
            '00:19:28',
            '00:19:99',
            '00:1A:D0',
            '00:1A:E8',
            '00:1B:1B',
            '00:1C:06',
            '00:30:05',
            '00:50:07',
            '00:90:40',
            '00:C0:E4',
            '08:00:06',
            ],
        'Samsung' : [
            '00:00:F0',
            '00:02:78',
            '00:09:18',
            '00:0D:AE',
            '00:0D:E5',
            '00:0F:73',
            '00:12:47',
            '00:12:FB',
            '00:13:77',
            '00:15:99',
            '00:15:B9',
            '00:16:32',
            '00:16:6B',
            '00:16:6C',
            '00:16:DB',
            '00:17:C9',
            '00:17:D5',
            '00:18:AF',
            '00:1A:8A',
            '00:1B:98',
            '00:1C:43',
            '00:1D:25',
            '00:1D:F6',
            '00:1E:7D',
            '00:E0:64',
            ],
        'LG' : [
            '00:05:C9',
            '00:0B:29',
            '00:12:56',
            '00:14:80',
            '00:19:A1',
            '00:1C:62',
            '00:1E:75',
            '00:1E:B2',
            '00:50:CE',
            '00:E0:91',
            ],
        'BenQ' : [
            '00:03:9D',
            '00:17:CA',
            ],
        'Motorola' : [
            '00:01:AF',
            '00:04:56',
            '00:04:BD',
            '00:08:0E',
            '00:0A:28',
            '00:0B:06',
            '00:0C:E5',
            '00:0E:5C',
            '00:0E:C7',
            '00:0F:9F',
            '00:11:1A',
            '00:11:80',
            '00:11:AE',
            '00:12:25',
            '00:12:8A',
            '00:12:C9',
            '00:13:71',
            '00:14:04',
            '00:14:9A',
            '00:14:E8',
            '00:15:2F',
            '00:15:9A',
            '00:15:A8',
            '00:16:26',
            '00:16:75',
            '00:16:B5',
            '00:17:00',
            '00:17:84',
            '00:17:E2',
            '00:17:EE',
            '00:18:A4',
            '00:18:C0',
            '00:19:2C',
            '00:19:5E',
            '00:19:A6',
            '00:19:C0',
            '00:1A:1B',
            '00:1A:66',
            '00:1A:77',
            '00:1A:AD',
            '00:1A:DB',
            '00:1A:DE',
            '00:1B:52',
            '00:1B:DD',
            '00:1C:11',
            '00:1C:12',
            '00:1C:C1',
            '00:1C:FB',
            '00:1D:6B',
            '00:1D:BE',
            '00:1E:46',
            '00:1E:5A',
            '00:1E:8D',
            '00:20:40',
            '00:20:75',
            '00:A0:BF',
            '00:C0:F9',
            '00:E0:0C',
            ],
        'Alcatel' : [
            '00:07:72',
            '00:08:9A',
            '00:0E:86',
            '00:0F:62',
            '00:11:3F',
            '00:11:8B',
            '00:15:3F',
            '00:16:4D',
            '00:17:CC',
            '00:19:8F',
            '00:1A:F0',
            '00:1C:8E',
            '00:1D:4C',
            '00:20:32',
            '00:20:60',
            '00:20:DA',
            '00:80:21',
            '00:80:39',
            '00:80:9F',
            '00:A0:81',
            '00:C0:BE',
            '00:D0:95',
            '00:D0:F6',
            '00:E0:B1',
            '00:E0:DA',
            ],
        }

